tests/test_libappimage.cpp

```diff
@@ -62,10 +62,22 @@ TEST_F(LibAppImageTest, appimage_get_type_invalid) {
     ASSERT_EQ(appimage_get_type("/tmp", false), -1);
 }
 
+TEST_F(LibAppImageTest, appimage_get_type_on_bare_iso_9660_file) {
+    ASSERT_EQ(appimage_get_type(iso_9660_file_path.c_str(), false), -1);
+}
+
+TEST_F(LibAppImageTest, appimage_get_type_on_bare_elf_file) {
+    ASSERT_EQ(appimage_get_type(elf_file_path.c_str(), false), -1);
+}
+
 TEST_F(LibAppImageTest, appimage_get_type_1) {
     ASSERT_EQ(appimage_get_type(appImage_type_1_file_path.c_str(), false), 1);
 }
 
+TEST_F(LibAppImageTest, appimage_get_type_on_appimage_type_1_withouth_magic_bytes) {
+    ASSERT_EQ(appimage_get_type(appImage_type_1_no_magic_file_path.c_str(), false), 1);
```

Comment:
You should rather unit-test the functions directly instead of bloating the test data directory...

Reply:
It's required to test the whole use case. Yes, the new function should be also tested separately.

Reply:
A unit test would cover the "has ISO but not ELF magic bytes", though. We should generate coverage reports.

Reply:
actually, there are tests for both cases, just add them see:

https://github.com/AppImage/libappimage/pull/12/files#diff-898c4b4229e6a1a0e75f8bba63b442c7R65

https://github.com/AppImage/libappimage/pull/12/files#diff-898c4b4229e6a1a0e75f8bba63b442c7R69

Reply:
I mean, we don't need a "real" ISO file to cover that code. In terms of line/branch coverage, that isn't necessary at all.

Reply:
The idea is to create a realistic test scenario.
Suggestions are welcome.

Reply:
There is no space for realism in TDD. The goal is to execute all lines >= once. You test the "realistic" frontend by calling it once. These lines are therefore covered. So, no need to test the rest through that interface. It'd be better to check the other lines by calling that functions directly. That avoids the need for additional test data.

Reply:
As you said you need to try to cover the whole code, therefore you need test cases for it. Reduced test cases only allow to validate one portion of it, you need also to validate the joints of the different modules. Therefore you need realistic test cases.
The function that is on discussion actually reads an ISO 9660 file so we must provide one, to be sure that it's done properly.

Reply:
@azubieta that doesn't make sense. Do you know what line coverage is?

Reply:
Re. the 300 kB: the more files are added, the more useless data is on my HDD. This could've been avoided. We need to set up coverage reports.

Reply:
I guess you mean [Code Coverage](https://en.wikipedia.org/wiki/Code_coverage) and if you add that you will realize that this part of the code wasn't covered by previous functions.

But again, if you came up with a way of improving this specific test please let me know.

Reply:
@azubieta the point is that in terms of line coverage, only one call to the `appimage_*` function would suffice to cover all lines in there. For the ISO specific bits, the left over lines can be tested separately. In terms of coverage, that's sufficiently tested then.

Line coverage is a special model of code coverage for source code that is mostly used as a synonym. Line coverage refers to lines in the source code that are executed.